generated/niscope/niscope/session.py

```diff
@@ -4729,6 +4779,64 @@ def export_attribute_configuration_file(self, file_path):
         errors.handle_error(self, error_code, ignore_warnings=False, is_error_handling=False)
         return
 
+    @ivi_synchronized
+    def get_ext_cal_last_date_and_time(self):
+        '''get_ext_cal_last_date_and_time
+
+        Returns the date and time of the last external calibration performed.
+
+        Returns:
+            last_cal_datetime (hightime.datetime): Indicates the **date** of the last calibration. A hightime.datetime object is returned, but only contains precision to the day.
+
+        '''
+
+        year, month, day = self._cal_fetch_date(enums._CalibrationTypes.EXTERNAL)
+        return hightime.datetime(year, month, day)
+
+    @ivi_synchronized
+    def get_ext_cal_last_temp(self):
+        '''get_ext_cal_last_temp
+
+        Returns the onboard temperature, in degrees Celsius, of an SMC-based oscilloscope at the time of the last successful external calibration.
+        The temperature returned by this node is an onboard temperature read from a sensor on the surface of the oscilloscope. This temperature should not be confused with the environmental temperature of the oscilloscope surroundings. During operation, the onboard temperature is normally higher than the environmental temperature.
+        Temperature-sensitive parameters are calibrated during self-calibration. Therefore, the self-calibration temperature is usually the more important temperature to read.
+
+        Returns:
+            temperature (float): Returns the **temperature** in degrees Celsius during the last calibration.
+
+        '''
+
+        return self._cal_fetch_temperature(enums._CalibrationTypes.EXTERNAL.value)
+
+    @ivi_synchronized
+    def get_self_cal_last_date_and_time(self):
+        '''get_self_cal_last_date_and_time
+
+        Returns the date and time of the last self calibration performed.
+
+        Returns:
+            last_cal_datetime (hightime.datetime): Indicates the **date** of the last calibration. A hightime.datetime object is returned, but only contains precision to the day.
+
+        '''
+
+        year, month, day = self._cal_fetch_date(enums._CalibrationTypes.SELF)
+        return hightime.datetime(year, month, day)
+
+    @ivi_synchronized
+    def get_self_cal_last_temp(self):
+        '''get_self_cal_last_temp
+
+        Returns the onboard temperature, in degrees Celsius, of an SMC-based oscilloscope at the time of the last successful self calibration.
+        The temperature returned by this node is an onboard temperature read from a sensor on the surface of the oscilloscope. This temperature should not be confused with the environmental temperature of the oscilloscope surroundings. During operation, the onboard temperature is normally higher than the environmental temperature.
+        Temperature-sensitive parameters are calibrated during self-calibration. Therefore, the self-calibration temperature is usually the more important temperature to read.
```

Comment:
more important compared to what?

Reply:
The external calibration temperature. This is sourced verbatim from official documentation of niScope_CalFetchTemperature.

Reply:
Since we're splitting `niScope_CalFetchTemperature` into two methods, we need to tweak the docstring a bit.


Maybe something like:
```
Therefore, the self-calibration temperature is usually more important than external calibration temperature.
```

Reply:
Makes sense. The documentation clarification has been made.

Reply:
>  of an SMC-based oscilloscope

If we are tweaking the documentation for our own wrappers then you should remove this obsolete reference to a specific hardware platform. There are newer non-SMC-based oscilloscopes that support storing the temperature that have been released since.

Reply:
Good to know. It's been changed.